src/core/resources/transactions/transaction.ts

```diff
@@ -218,32 +189,53 @@ export const useTransaction = ({
   chainId?: number;
   hash?: `0x${string}`;
 }) => {
-  const customChain = !!chainId && isCustomChain(chainId);
-  const {
-    data: backendTransaction,
-    isLoading: backendTransactionIsLoading,
-    isFetched: backendTransactionIsFetched,
-  } = useBackendTransaction({
-    hash,
-    chainId,
-    enabled: !customChain && !!hash && !!chainId,
+  const queryClient = useQueryClient();
+
+  const { currentAddress: address } = useCurrentAddressStore();
+  const { currentCurrency: currency } = useCurrentCurrencyStore();
+  const { chains } = useUserChains();
+
+  const params = {
+    hash: hash!,
+    address,
+    currency,
+    chainId: chainId!,
+  };
+
+  const consolidatedTransactionsKey = consolidatedTransactionsQueryKey({
+    address,
+    currency,
+    userChainIds: chains.map((chain) => chain.id),
   });
-  const {
-    data: providerTransaction,
-    isLoading: providerTransactionIsLoading,
-    isFetched: providerTransactionIsFetched,
-  } = useCustomNetworkTransaction({
-    hash,
-    chainId,
-    enabled: customChain,
+
+  return useQuery({
+    queryKey: createQueryKey('transaction', params),
+    queryFn: () => fetchTransaction(params),
+    enabled: !!hash && !!address && !!chainId,
+    initialData: () => {
+      // consolidatedTransactions is only for BE supported chains
+      const queryData = queryClient.getQueryData<PaginatedTransactions>(
+        consolidatedTransactionsKey,
+      );
+      const pages = queryData?.pages || [];
+      for (const page of pages) {
+        const tx = page.transactions.find(
+          (tx) => tx.hash === hash && tx.chainId === chainId,
+        );
+        if (tx) return tx;
+      }
+
+      // try to find in custom network transactions store
+      const { getCustomNetworkTransactions } =
+        customNetworkTransactionsStore.getState();
+      return getCustomNetworkTransactions({ address }).find(
+        (tx) => tx.hash === hash && tx.chainId === chainId,
+      );
+    },
```

Comment:
i don't follow why this is mixing txs supported by backend and the ones from `customNetworkTransactionsStore`, there are separate stores and we're using them separately